tests/integration/testIntegration.nim

```diff
@@ -134,3 +134,27 @@ twonodessuite "Integration tests", debug1 = false, debug2 = false:
     await provider.advanceTime(duration)
 
     check eventually (await token.balanceOf(account2)) - startBalance == duration*reward
+
+  test "expired request partially pays out for stored time":
```

Comment:
This test is failing in CI?

Reply:
Yes as per the comment in the PR description:

> The integration test is failing, though, because of the Hardhat's mining mode - automine. The time needs to be advanced (eq. mined block) before calling the withdrawFunds() function for the canceled request, but I have not figured out any way how to do it in the test's code. I have tried switching the Hardhat's mining mode into mining block every second and for that the test worked. Maybe we should consider changing the mining mode? It had brought us more problems recently 😳

Do you have any thoughts? If you don't have any ideas on how to accomplish this, nor you are against changing the Hardhat's mining mode, then I will drop (comment out) the test for it to pass...

Reply:
Of course, sorry :facepalm:

You can advance the time in the integration tests as well using:
```nim
    await provider.advanceTimeTo(expiry)
```

But it seems that the test is failing earlier; the requestStorage() call fails.

Changing the mining mode has a couple of problems that I can see:
- the tests take longer to run if you have to wait a second for a tx to be mined
- most tests would need to be rewritten because they assume manual mining
- tests might become brittle because of the timing differences when things are mined

Reply:
> You can advance the time in the integration tests as well using:

I am aware of this, but the problem is to call this at the right moment. It needs to be called once the state machines (both for seller and buyer) reach `Cancelled` state, which I don't know how to do that. I could either add some callback which would be called in `Cancelled` state (that would not actually much help here, as this is integration tests), I could try to time it (don't think that is good idea).

Or we could simply drop this test, because I have similar one ready in the Dist-testing framework: https://github.com/codex-storage/cs-codex-dist-tests/pull/55

Reply:
> It needs to be called once the state machines (both for seller and buyer) reach `Cancelled` state

Why do we need to advance the time after the request is cancelled on chain? (I'm probably missing something obvious here)

In thought it should work as follows:
1. client requests storage on 2 nodes
2. only 1 node fills a slot
3. the request expires (initiated by the advanceTime() call)
4. both client and the node that filled a slot withdraw their funds